forge/ee/routes/pipeline/index.js

```diff
@@ -438,7 +438,35 @@ module.exports = async function (app) {
      * Approach depends on stage.action
      */
     app.put('/pipelines/:pipelineId/stages/:stageId/deploy', {
-        preHandler: app.needsPermission('pipeline:edit')
+        preHandler: app.needsPermission('pipeline:edit'),
+        schema: {
+            summary: 'Triggers a pipeline stage',
+            tags: ['Pipelines'],
+            params: {
+                type: 'object',
+                properties: {
+                    pipelineId: { type: 'string' },
+                    stageId: { type: 'string' }
+                }
+            },
+            body: {
+                type: 'object',
+                properties: {
+                    sourceSnapshotId: { type: 'string', description: 'The snapshot to deploy if the stage action is set to "prompt"' }
```

Comment:
This is optional, does it need to be flagged as such?

Reply:
Properties are optional by default. Otherwise, we would set the `required` flag on them